### PR TITLE
Remove dependency from ocamlformat to ocamlformat-lib

### PR DESCRIPTION
**Pull request by gpetiot**
Follow up on #2230

### CODE REVIEW

**Review by Julow — COMMENTED** [review]
This doesn't work, `dune build -p ocamlformat`:
```
File "bin/ocamlformat/dune", line 21, characters 12-27:
21 |  (libraries ocamlformat-lib bin_conf))
                 ^^^^^^^^^^^^^^^
Error: Library "ocamlformat-lib" not found.
```

This seems like it should be possible but I didn't manage to fix it.

**Comment by gpetiot** [comment]
> 

This works for me on this branch.

**Comment by gpetiot** [comment]
I tried to improve the situation with `copy_files` in the dune files without any success, for reference the last error is:
```
File "lib/bin_conf/dune", line 14, characters 41-75:
14 |  (libraries ocamlformat-lib.ocaml_common ocamlformat-lib.ocamlformat_stdlib re))
                                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Error: Library "ocamlformat-lib.ocamlformat_stdlib" not found.
-> required by library "ocamlformat.bin_conf" in _build/default/lib/bin_conf
etc.
```
The 2 libs mentioned are private, but using `copy_files` to get their sources instead doesn't fix the issue.

The combination of private and public libraries makes it tricky. I'm out of ideas for now.

File: vendored-lib/dune [hunk 1]
```diff
@@ -0,0 +1,22 @@
+(library
+ (name vendored_lib)
+ (public_name ocamlformat.vendored_lib)
```
Comment (kit-ty-kate) [inline vendored-lib/dune:3]: I'm not sure to understand the issue, why exporting the vendored library?
```suggestion
```

**Comment by gpetiot** [comment]
I've played around with different solutions without success.  At this point I'm wondering if this wouldn't be better for someone else to just take over this PR (free hand :D), possibly sanitize the whole situation with vendored libs, we probably don't follow `dune`'s best practices.

We don't have a lot of constraints regarding which libs should be private or public as long as we abide by the existing opam files.

For context that's the last issue I wanted to fix before wrapping up the release.

**Comment by gpetiot** [comment]
After discussing with @emillon we cannot do better than what is already done on `main`. Closing this one.

**Pull request closed without merging**